### PR TITLE
add grpc services.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -1,72 +1,81 @@
...
+    uint64 size = 2;
+    uint32 checksum = 3;
 }
 
 message SnapshotMeta {
     repeated SnapshotCFFile cf_files = 1;
 }
 
+message SnapshotChunk {
+    RaftMessage message = 1;
```
Comment (siddontang) [inline proto/raft_serverpb.proto]: how can we check this is the first or last snapshot chunk?

Comment (disksing) [inline proto/raft_serverpb.proto]: `if has_message() {/* first */} else if has_data() {/* rest */} else { /* error */ }`

File: Makefile [hunk 2]
```diff
@@ -10,7 +10,6 @@ export GOPATH := $(GOPATH):$(CURDIR)/_vendor
 all: go rust test
 
 test:
```
Comment (hhkbp2) [inline Makefile]: Remove target `test` if there is no actions to be taken for it.

File: proto/raft_serverpb.proto (hunk 1 above)
Comment (hhkbp2) [inline proto/raft_serverpb.proto]: Some comment on this will be helpful.

**Comment by hhkbp2** [comment]
Rest LGTM

File: proto/raft_serverpb.proto [hunk 3]
```diff
@@ -1,72 +1,82 @@
...
+    string cf = 1;
+    uint64 size = 2;
+    uint32 checksum = 3;
 }
 
 message SnapshotMeta {
     repeated SnapshotCFFile cf_files = 1;
 }
 
+// Snapshot is sent by chunks, the first chunk contains the RaftMessage, following chunks contain only data.
```
Comment (siddontang) [inline proto/raft_serverpb.proto]: how about the last chunk?

Comment (disksing) [inline proto/raft_serverpb.proto]: Snapshot will close the stream after sending the last chunk.

**Comment by disksing** [comment]
PTAL @siddontang @hhkbp2

**Comment by siddontang** [comment]
Do we have any test in this repository?

**Comment by disksing** [comment]
@siddontang No tests.

**Comment by siddontang** [comment]
Maybe it is better to remove the vendor, I am very confused with this.

**Comment by disksing** [comment]
@siddontang The vendor dir is not introduced by this PR. I only add the grpc dependency.

**Comment by siddontang** [comment]
Seem not used, we can remove it later.

**Comment by disksing** [comment]
The vendor can help to make sure upstream's changes won't break our CI.

**Comment by siddontang** [comment]
why import tipb script in vendor?

File: Cargo.toml [hunk 4]
```diff
@@ -12,4 +12,5 @@ futures = "0.1"
 futures-cpupool = "0.1"
 
 [dependencies.grpc]
-git = "https://github.com/stepancheg/grpc-rust.git"
+git = "https://github.com/pingcap/grpc-rs.git"
+branch = "busyjay/dev"
```
Comment (siddontang) [inline Cargo.toml]: should use static link here?

**Comment by siddontang** [comment]
Rest LGTM

File: Cargo.toml (hunk 4 above)
Comment (disksing) [inline Cargo.toml]: Later. This branch doesn't have static link feature now.

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**